### DISCUSSION TITLE
Snapshot plugin failures on all cameras (Hikvision and Amcrest)

### DISCUSSION THREAD

**Discussion by SirBigBits** [koush/scrypted#705] [Q&A]
[Snapshot Plugin]: Snapshot failed RPCResultError: Request failed with status code 404
[Snapshot Plugin]:     at createError (/node_modules/axios/lib/core/createError.js:16:15)
[Snapshot Plugin]:     at settle (/node_modules/axios/lib/core/settle.js:17:12)
[Snapshot Plugin]:     at IncomingMessage.<anonymous> (/node_modules/axios/lib/adapters/http.js:269:11)
[Snapshot Plugin]:     at IncomingMessage.emit (node:events:525:35)
[Snapshot Plugin]:     at IncomingMessage.emit (node:domain:489:12)
[Snapshot Plugin]:     at endReadableNT (node:internal/streams/readable:1359:12)
[Snapshot Plugin]:     at processTicksAndRejections (node:internal/process/task_queues:82:21)
[Snapshot Plugin]: @scrypted/amcrest:host
[Snapshot Plugin]: host:@scrypted/snapshot {
[Snapshot Plugin]:   cause: undefined
[Snapshot Plugin]: }
[Snapshot Plugin]: creating error image with background Snapshot Failed true
listen loop connection failed, restarting listener. Request failed with status code 404
[Snapshot Plugin]: Snapshot failed RPCResultError: Request failed with status code 404
[Snapshot Plugin]:     at createError (/node_modules/axios/lib/core/createError.js:16:15)
[Snapshot Plugin]:     at settle (/node_modules/axios/lib/core/settle.js:17:12)
[Snapshot Plugin]:     at IncomingMessage.<anonymous> (/node_modules/axios/lib/adapters/http.js:269:11)
[Snapshot Plugin]:     at IncomingMessage.emit (node:events:525:35)
[Snapshot Plugin]:     at IncomingMessage.emit (node:domain:489:12)
[Snapshot Plugin]:     at endReadableNT (node:internal/streams/readable:1359:12)
[Snapshot Plugin]:     at processTicksAndRejections (node:internal/process/task_queues:82:21)
[Snapshot Plugin]: @scrypted/amcrest:host
[Snapshot Plugin]: host:@scrypted/snapshot {
[Snapshot Plugin]:   cause: undefined
[Snapshot Plugin]: }
[Snapshot Plugin]: creating error image with background Snapshot Failed true
listen loop connection failed, restarting listener. Request failed with status code 404
listen loop connection failed, restarting listener. Request failed with status code 404

**Comment by SirBigBits** [comment]
Is there a confirmed Snapshot Plugin issue or is it just my setup?  I've done 3 fresh re-installs on 2 different machines with the same issue.

**Reply by Dudleydogg** [reply]
NO i have exact same issue, the Hikvision plug-in video works very well but without the snapshots it does not fit my needs. Really need the snapshots, currently they only update after closing a feed, but in the Scrypted portal I see the snapshot just never makes it to home kit, If I could find a fix for this I would move everything from FFMPG to hikvision.

**Reply by Dudleydogg** [reply]
I built a new one, INstalled NO plugins except HomeKit and Hikvision, Snapshots all started working, then had to do Transcode and WebRTC and everything works, pulling Stream directly from NVR and every 10 seconds a snapshot pulls in. There were other plugins conflicting, after removing all the plugs but just those few it was working now still testing, but wondering now if my ring door bell may work and I gave up too soon.

**Reply by Adam-Langley** [reply]
I have Hikvision cameras, and noticed that HSV motion events stopped working today after I had been reconfiguring a camera.
I also noticed the "listen loop connection failed, restarting listener." Mine had a different HTTP status code however - "listen loop connection failed, restarting listener. Request failed with status code 401"

I resolved it by turning something off that I had turned on:

<img width="451" alt="Screenshot 2023-11-09 at 5 30 46 PM" src="https://github.com/koush/scrypted/assets/16373742/5a08c4ac-b001-4a87-976c-dc827aa1f4fa">
